crates/api_crud/src/multi_community/update.rs

```diff
@@ -0,0 +1,38 @@
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
+use chrono::Utc;
+use lemmy_api_common::{
+  community::UpdateMultiCommunity,
+  context::LemmyContext,
+  LemmyErrorType,
+  SuccessResponse,
+};
+use lemmy_db_schema::{
+  source::multi_community::{MultiCommunity, MultiCommunityUpdateForm},
+  traits::Crud,
+  utils::diesel_string_update,
+};
+use lemmy_db_views_community::{multi_community::ReadParams, MultiCommunityView};
+use lemmy_db_views_local_user::LocalUserView;
+use lemmy_utils::error::LemmyResult;
+
+pub async fn update_multi_community(
+  data: Json<UpdateMultiCommunity>,
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<SuccessResponse>> {
+  // check that owner is correct
+  let read = MultiCommunityView::read(&mut context.pool(), ReadParams::Id(data.id)).await?;
+  if read.multi.creator_id != local_user_view.person.id {
+    return Err(LemmyErrorType::MultiCommunityUpdateWrongUser.into());
+  }
```

Comment:
You are doing this in multiple places, so might as well move it to a common function.

Reply:
Done